opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -431,6 +441,134 @@ message HistogramDataPoint {
+  fixed64 start_time_unix_nano = 2;
+
+  // TimeUnixNano is required, see the detailed comments above Metric.
+  //
+  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January
+  // 1970.
+  fixed64 time_unix_nano = 3;
+
+  // count is the number of values in the population. Must be non-negative. This
+  // value must be equal to the sum of the "count" fields in buckets if a
+  // histogram is provided.
+  fixed64 count = 4;
+
+  // sum of the values in the population. If count is zero then this field
+  // must be zero. This value must be equal to the sum of the "sum" fields in
+  // buckets if a histogram is provided.
+  //
+  // Note: Sum should only be filled out when measuring non-negative discrete
+  // events, and is assumed to be monotonic over the values of these events.
+  // Negative events *can* be recorded, but sum should not be filled out when
+  // doing so.  This is specifically to enforce compatibility w/ OpenMetrics,
+  // see: https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#histogram
+  double sum = 5;
+
+  // scale describes the resolution of the histogram.  Boundaries are
+  // located at powers of the base, where:
+  //
+  //   base = (2^(2^-scale))
+  //
+  // The histogram bucket identified by `index`, a signed integer,
+  // contains values that are less than or equal to (base^index) and
+  // greater than (base^(index-1)).
+  //
+  // The positive and negative ranges of the histogram are expressed
+  // separately but both use the same scale to locate their boundaries.
+  //
+  // scale is valid in the range -4 through +8 inclusive.
+  sint32 scale = 6;
+
+  // zero_count is the count of values that are either exactly zero or
+  // within the region considered zero by the instrumentation at the
+  // tolerated degree of precision.  This bucket stores values that
+  // cannot be expressed using the standard exponential formula as
+  // well as values that have been rounded to zero.  Users have the
+  // option to set zero_tolerance to convey additional information about
+  // the width of the zero region.
+  //
+  // Implementations MAY consider the zero bucket to have probability
+  // mass equal to (zero_count / sum).
+  fixed64 zero_count = 7;
+
+  // zero_tolerance may be optionally set to convey the width of the
+  // zero region.  When this is not set, implementations may assume
+  // the zero tolerance equals the smallest representable positive
+  // IEEE 754 double-precision floating point number, which is (2^−1074).
+  //
+  // Producers can be configured to round measurements to zero that fall
+  // below the zero tolerance.  Users should avoid producing measurements
+  // that have meaningless precision near zero, as exponential buckets
+  // become increasingly dense in this region.
+  //
+  // Implementations MAY consider the zero bucket to have probability
+  // density equal to (zero_count / 2*zero_tolerance), if set.
+  double zero_tolerance = 8;
```

Comment:
I am also not sure if the `zero_tolerance` field is really needed. However, if it is needed, I would rather define it as the largest nonnegative double-precision value that maps to the zero bucket. (Currently, it is defined as the smallest nonnegative value that maps to a regular bucket. ) As a consequence, the default value would be 0 and not the "magic number" `2^-1074`.

This definition would also be more consistent as we would have exclusive lower bounds for all buckets. Consider the first regular bucket to which values may be mapped to. With the proposed definition of `zero_tolerance`, this bucket would contain all values from the interval `(zero_tolerance, base^i]` where `i` is the bucket index. Hence, we would have again an exclusive lower bound as for all other buckets which contain values from `(base^(i-1), base^i]`.

Reply:
I like the suggestion by @oertl that zero_tolerance could be defined so that the zero value (default) is an inclusive upper bound.

I've removed this field from the protocol because it can be added later, by this logic, at which time the can of worms described by @yzhuge will have to be addressed. To add this field we will need to pin down more detail about how to merge histograms with different zero tolerances, for example.